synapse/handlers/groups_local.py

```diff
@@ -365,6 +365,32 @@ async def join_group(self, group_id, user_id, content):
 
         return {}
 
+    async def force_join_user_to_group(self, group_id, user_id):
+        """Forces a user to join a group.
+        """
+        if not self.is_mine_id(group_id):
+            raise SynapseError(400, "Can only affect local groups")
+
+        if not self.is_mine_id(user_id):
+            raise SynapseError(400, "Can only affect local users")
+
+        # Bypass the group server to avoid business logic regarding whether or not
+        # the user can actually join.
+        await self.store.add_user_to_group(group_id, user_id)
```

Comment:
In some ways this belongs in `groups_server.py`, but its probably fine to ignore the split between the two files in this particular instance.